W-13114428-rtf-helm-migration-kt (#1086)
mulesoft/docs-runtime-fabric: modules/ROOT/pages/migrate-helm.adoc

== Migrating to Helm Managed Runtime Fabric

You can migrate to xref:runtime-fabric::install-helm.adoc[helm] native for Runtime Fabrics that were installed using xref:runtime-fabric::install-self-managed.adoc[rtfctl].

== Before You Begin

To perform the migration, you must consider the following:

* The migration is applicable to BYOK clusters which are installed with `rtfctl` and want to migrate to helm managed.

> * The cluster must be on `rtf-agent` v2.2.4 or above. Note that you will first have to upgrade latest before you can initiate the migration. 

Reviewer: ```suggestion
* The cluster must be on `rtf-agent` v2.2.5 or above. Note that you will first have to upgrade agent to the required version before you can initiate the migration. 
```

Additionally, you must have these details:

* organizationId
* fabricId
* Bearer token for Anypoint user account with *manage fabric* privileges
* rtfctl version >= v1.0.10 is downloaded

== Migrate rtfctl to Helm

To migrate your Runtime Fabric from rtfctl to helm, run this command:

[source,copy]
----
rtfctl migrate previous-rtf-bkp.tar.gz --organizationId <orgId value> --fabricId <fabricId value> --token <bearer token> --host qax.anypoint.mulesoft.com  --confirm
----

Running this command performs the following actions:

* Backs up the current Runtime Fabric installation: tarball is created with input file name.
* Uninstalls the current Runtime Fabric installation: running the command does not update Mule application deployments, it does not delete any of your configurations applied to the rtf namespace, it does not delete the rtf namespace, and it does not touch any other namespace than the rtf namespace. Uninstalls only the previous runtime-fabric helm chart.
* Regenerates activation data for provided fabricId.

> * Deletes custom-properties secret in the rtf namespace, because in helm managed rtf the secret is recreated using values from the helm chart.

Reviewer 2: ```suggestion
* Deletes custom-properties secret in the rtf namespace, because helm managed rtf recreates the secret using values from the helm chart.
```

> * Installs Runtime Fabric with helm commands, such as re-rendering Runtime Fabric helm chart.

Reviewer: ```suggestion
* Installs Runtime Fabric with helm commands, to re-render Runtime Fabric helm chart.
```

> * Installs Runtime Fabric with helm commands (e.g. re-renders Runtime Fabric helm chart).

Reviewer 2: ```suggestion
* Installs Runtime Fabric with helm commands, such as re-rendering Runtime Fabric helm chart.
```

> Running the command does not upgrade Runtime Fabric to the other version than the existing installation version.

Reviewer 2: ```suggestion
Running the command does not upgrade Runtime Fabric to the version other than the existing installation version.
```

=== Command Success

> If running the command is successfull:

Reviewer: ```suggestion
If running the command is successful user can:
```

Reviewer: If you want to keep this line `If running the command is successful:`  it has a typo in `successful`
Also, then it makes  
`Manage (start, stop, delete, update) previously deployed Mule applications from the Runtime Manager UI.
Deploy Mule applications to the target Runtime Fabric from the Runtime Manager UI.`
These two lines confusing as it is not necessary to do those things after command is successful. 
On the other hand even
`Verify that Runtime Fabric is now helm managed in the Anypoint Runtime Manager UI.`
and. 
`Verify that configurations such as ingress templates, persistence gateway, custom resources and proxy values that you applied before the migration are intact.` are not required. 

You can either specify that user can do these things or just remove this paragraph.

Reviewer: @kevintroller can you please fix the typo in `successfull` ?

> * Verify that Runtime Fabric is now helm managed either in the Anypoint Runtime Manager UI or via helm values.

Reviewer: ```suggestion
* Verify that Runtime Fabric is now helm managed in the Anypoint Runtime Manager UI.
```

Reviewer: @kevintroller checking that from helm values is hard and confusing to user.

* Manage (start, stop, delete, update) previously deployed Mule applications from the Runtime Manager UI.
* Deploy Mule applications to the target Runtime Fabric from the Runtime Manager UI.

> * Configurations such as ingress templates, persistence gateway, custom resources and proxy values that you applied before the migration are intact.

Reviewer: ```suggestion
* Verify that configurations such as ingress templates, persistence gateway, custom resources and proxy values that you applied before the migration are intact.
```
